### PR TITLE
[RemoveDIs] Update syntax highlighting to include debug records

### PR DESCRIPTION
**Pull request by SLTozer**
This patch updates the vim and vscode syntax highlighters to recognize debug records. I wasn't able to build the vscode extensions locally, so I've only tested this pattern by checking that the pattern works in vscode search.

### CODE REVIEW

**Review by OCHyams — COMMENTED** [review]
Is there someone else you can add to the review that has edited the file or reviewed a recent edit?

Rather than using "fancy" regex I would rather just have `#dbg_assign` `#dbg_label` etc checked separately. Less error prone, easier to read (for me!), but I won't push strongly for it.

Any chance we can get emacs too (https://github.com/llvm/llvm-project/blob/main/llvm/utils/emacs/llvm-mode.el) - I can look at adding later if you're not sure about emacs.

File: llvm/utils/vim/syntax/llvm.vim [hunk 1]
```diff
@@ -36,6 +36,9 @@ syn keyword llvmStatement sle slt srem store sub switch trunc udiv ueq uge ugt
 syn keyword llvmStatement uitofp ule ult umax umin une uno unreachable unwind
 syn keyword llvmStatement urem va_arg xchg xor zext
 
+" Debug records.
+syn match llvmStatement /\v#dbg_(assign|declare|label|value)/
```
[the remark this answers is not in the document]
Comment (felipepiovezan) [reply llvm/utils/vim/syntax/llvm.vim:40]: `/` is just what you type in vim when you're about to start a search
`\v` is to control the "very magic" behaviour of regex spelling:

> Vim's default 'magic' setting makes characters have the same meaning as in grep, and \v (very magic) makes them the same as the extended regular expressions used by egrep.

**Review by felipepiovezan — APPROVED** [review]
vim changes LGTM

**Review by jryans — APPROVED** [review]
VS Code changes look good to me.

**Comment by OCHyams** [comment]
Tested the emacs change locally, that LGTM (not super familiar, but it works and looks reasonable).

**Pull request merged**